client/gutenberg/extensions/related-posts/edit.jsx

```diff
@@ -0,0 +1,123 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { __ } from '@wordpress/i18n';
+import { Fragment } from '@wordpress/element';
+import { BlockControls, InspectorControls } from '@wordpress/editor';
+import {
+	Button,
+	PanelBody,
+	RangeControl,
+	TextControl,
+	ToggleControl,
+	Toolbar,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { DEFAULT_POSTS, MAX_POSTS_TO_SHOW } from './constants';
+
+export default ( { attributes, className, setAttributes } ) => {
+	const {
+		displayContext,
+		displayDate,
+		displayThumbnails,
+		headline,
+		postLayout,
+		postsToShow,
+	} = attributes;
+
+	const layoutControls = [
+		{
+			icon: 'grid-view',
+			title: __( 'Grid View' ),
+			onClick: () => setAttributes( { postLayout: 'grid' } ),
+			isActive: postLayout === 'grid',
+		},
+		{
+			icon: 'list-view',
+			title: __( 'List View' ),
+			onClick: () => setAttributes( { postLayout: 'list' } ),
+			isActive: postLayout === 'list',
+		},
+	];
+
+	const displayPosts =
+		DEFAULT_POSTS.length > postsToShow ? DEFAULT_POSTS.slice( 0, postsToShow ) : DEFAULT_POSTS;
```

Comment:
Fun fact! This should be equivalent to the inner case…

```js
const displayPosts = DEFAULT_POSTS.slice( 0, postToShow );
```

the only difference is that we are cloning the array each time and as-written we're not cloning when the array is empty. since this is _inside_ a `render()` function it won't trigger additional re-renders

unless there's a reason to leave it here we should be able to remove the conditional altogether.

Reply:
Good point 👍 Done in https://github.com/Automattic/wp-calypso/pull/26530/commits/be96069c0d3213b30f636eeb78292be454407928.